### PR TITLE
Data 4407 update current event in merge

### PR DESCRIPTION
**Pull request by liatiusim**

### CODE REVIEW

File: plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/MergeProfilesOnPropertyAction.java [hunk 1]
```diff
@@ -182,6 +182,11 @@ public int execute(Action action, Event event) {
                     @Override
                     public boolean execute() {
                         try {
+                            // Update current event explicitly, as it might not return from search query if there wasn't a refresh in ES
+                            if (!StringUtils.equals(profileId, masterProfileId)) {
```
Comment (giladw) [inline plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/MergeProfilesOnPropertyAction.java:186]: (!StringUtils.equals(profileId, masterProfileId) can we make it  more clear?
which profileId is it. the event's?
on line 163 there is  event.setProfileId(masterProfileId);

Comment (liatiusim) [reply plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/MergeProfilesOnPropertyAction.java:186]: yes, it's the event id.
The set in line 163 is not saved in elastic, this is why we need yo update here directly.
I only read existing variables, didn't define them. Do you have an idea how can it be more clear?

Comment (giladw) [reply plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/MergeProfilesOnPropertyAction.java:186]: who the profileId suppose to be? maybe we can use event.profileId?

Comment (liatiusim) [reply plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/MergeProfilesOnPropertyAction.java:186]: Just used an existing variable.. the same one which is in use in the whole action and later in line 192 - and better to keep the convention of using it. We can have a look over the whole action and see if the variables have proper names - but maybe in a different PR?

**Review by shirbr — APPROVED** [review]

**Pull request merged**